src/compiler/parser.ts

```diff
@@ -4737,6 +4737,14 @@ namespace ts {
          *
          */
         function parseUnaryExpressionOrHigher(): UnaryExpression | BinaryExpression {
+            /**
+             * If we have a PrivateIdentifier, parse this unconditionally.
+             * A privateIdentifier is only valid on its own in the RelationalExpression: `#field in expr`.
+             * The checker will emit an error if this is not the case.
+             */
+            if (token() === SyntaxKind.PrivateIdentifier) {

```

Comment:
Could this be in `parsePrimaryExpression` right above `parseIdentifier`? I'd be even more comfortable with it there.

Reply:
Moved this check down the parsing levels into `parsePrimaryExpression` and all OK. Much nicer, thanks!